### PR TITLE
Fetch: fragments

### PR DESCRIPTION
**Pull request by annevk**
Tests for https://github.com/whatwg/fetch/pull/696.

### CODE REVIEW

File: fetch/fragments/subresource-redirect.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+<img data-desc="Redirect with the original URL containing a fragment"
+     src=../api/resources/redirect.py?location=/images/green.svg#svgView(viewBox(25,25,50,50))>
+<img data-desc="Redirect with the response Location header containing a fragment"
+     src=../api/resources/redirect.py?location=/images/green.svg%23svgView(viewBox(25,25,50,50))>
+<img data-desc="Redirect with both the original URL and response Location header containing a fragment"
+     src=../api/resources/redirect.py?location=/images/green.svg%23svgView(viewBox(25,25,50,50))#svgView(viewBox(0,0,50,50))>
+<canvas width=100 height=50></canvas>
+<script>
+const ctx = document.querySelector("canvas").getContext("2d"),
+      isColor = (data, r, g, b, a) => {
```
Comment (annevk) [inline fetch/fragments/subresource-redirect.html:27]: This needs to become less precise somehow as it creates intermittent results in Safari. Probably borrow something from canvas tests?

**Comment by domenic** [comment]
I've been doing mostly editorial review for these changes and haven't been following the intended behavior, so I'll reassign this to @wanderview who seems to have been following more closely.

**Review by wanderview — APPROVED** [review]
LGTM.  @davidben might want to look as well.

**Review by davidben — APPROVED** [review]
LGTM. Given the cause of the issues in Chrome (SVG-specific URL plumbing silliness that's different in different contexts), we may extend these later, but no reason to not land the initial tests now.

(I suspect we'll want tests for both `<canvas>` and actually rendering. And maybe tests that redirect `color.svg#green` and `color.svg#red` in the same page, to capture potential caching mishaps.)

**Pull request merged**